src/components/behandlerdialog/meldingtilbehandler/VelgBehandler.tsx

```diff
@@ -1,53 +1,126 @@
-import { Field } from "react-final-form";
+import React, { useEffect, useState } from "react";
 import { BehandlerDTO } from "@/data/behandler/BehandlerDTO";
 import AppSpinner from "@/components/AppSpinner";
-import BehandlerRadioGruppe, {
-  BehandlerRadioGruppeTexts,
-} from "@/components/behandler/BehandlerRadioGruppe";
 import { useBehandlereQuery } from "@/data/behandler/behandlereQueryHooks";
+import { Radio, RadioGroup } from "@navikt/ds-react";
+import {
+  FieldErrors,
+  UseFormRegister,
+  UseFormSetError,
+  UseFormTrigger,
+  UseFormWatch,
+} from "react-hook-form";
+import { MeldingTilBehandlerSkjemaValues } from "@/components/behandlerdialog/meldingtilbehandler/MeldingTilBehandlerSkjema";
+import { capitalizeWord } from "@/utils/stringUtils";
+import BehandlerSearch from "@/components/behandler/BehandlerSearch";
 
-const behandlerRadioGruppeTexts: BehandlerRadioGruppeTexts = {
+const texts = {
   behandlerLegend: "Velg behandler som skal motta meldingen",
   behandlersokTekst: "Søk etter behandler",
+  missingBehandler: "Vennligst velg behandler",
 };
 
 interface VelgBehandlerProps {
   selectedBehandler: BehandlerDTO | undefined;
   setSelectedBehandler: (behandler?: BehandlerDTO) => void;
+  register: UseFormRegister<MeldingTilBehandlerSkjemaValues>;
+  watch: UseFormWatch<MeldingTilBehandlerSkjemaValues>;
+  setError: UseFormSetError<MeldingTilBehandlerSkjemaValues>;
+  trigger: UseFormTrigger<MeldingTilBehandlerSkjemaValues>;
+  errors: FieldErrors<MeldingTilBehandlerSkjemaValues>;
 }
 
 export const VelgBehandler = ({
   selectedBehandler,
   setSelectedBehandler,
+  register,
+  watch,
+  trigger,
+  setError,
+  errors,
 }: VelgBehandlerProps) => {
   const { data: behandlere, isInitialLoading } = useBehandlereQuery();
-  const field = "behandlerRef";
+  const [showBehandlerSearch, setShowBehandlerSearch] =
+    useState<boolean>(false);
+
+  const behandlerOneliner = (behandler: BehandlerDTO): string => {
+    const name = [behandler.fornavn, behandler.mellomnavn, behandler.etternavn]
+      .filter(Boolean)
+      .join(" ");
+    const type = !!behandler.type ? `${capitalizeWord(behandler.type)}:` : "";
+    const typeAndName = `${type} ${name}`;
+    const office = !!behandler.kontor ? capitalizeWord(behandler.kontor) : "";
+    const phone = !!behandler.telefon ? `tlf ${behandler.telefon}` : "";
+
+    return [typeAndName, office, phone].filter(Boolean).join(", ");
+  };
+
+  const behandlerRefField = register("behandlerRef", {
+    required: true,
+    validate: (value) => {
+      if (value === "sokEtterBehandler" && selectedBehandler === undefined) {
+        setError("behandlerRef", { type: "custom" });
+        return false;
+      }
+      return true;
+    },
+  });
+
+  const behandlerRefValue = watch("behandlerRef");
+  useEffect(() => {
+    if (behandlerRefValue === "sokEtterBehandler") {
+      trigger("behandlerRef");
+    }
+  }, [behandlerRefValue, selectedBehandler, trigger]);
```

Comment:
Når vi velger `sokEtterBehandler` må vi trigge (`trigger("behandlerRef")`) validering for `behandlerRef` feltet ettersom vi først må velge en behandler fra søket før radio button valget er gyldig.

Reply:
Ser at jeg får valideringsfeil "Vennligst velg behandler" med en gang jeg velger Søk behandler (uten å ha trykket Send).

Reply:
Sletter denne da 😁 Den ble lagt til for å få til akkurat dette 🙃